services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,19 +43,62 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) != pdPASS) {
+    return ERR_CODE_QUEUE_FULL;
+  }
+
+  return ERR_CODE_SUCCESS;
+}
+error_code_t thermalMgrSendEventToFront(thermal_mgr_event_t *event) {
+  if (xQueueSendToFront(thermalMgrQueueHandle, event, 0) != pdPASS) {
+    return ERR_CODE_QUEUE_FULL;
+  }
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+  thermal_mgr_event_t event = {
+    type: THERMAL_MGR_EVENT_CHECK_SAFE_TEMP_CMD
+  };
+  thermalMgrSendEventToFront(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  lm75bd_config_t config = *(lm75bd_config_t *) pvParameters;
+
   while (1) {
-    
+    thermal_mgr_event_t event;
+    if (xQueueReceive(thermalMgrQueueHandle, &event, 0) == pdPASS) {
```

Comment:
since this task doesnt do anything else unless an event is received, you shouild block inddefinetely (`portMAX_DELAY`) rather than polling the queue